connector/src/test/java/com/google/cloud/hive/bigquery/connector/IntegrationTests.java

```diff
@@ -157,4 +161,148 @@ public void testUnsupportedTypes() {
     }
     tearDown();
   }
+
```

Comment:
Notice that tearDown() won't be called on assertion failure. It is best to mark setup() and tearDown() with @Before and @After, and use the @RunWith(Parameterized.class) to provide the engines.

Reply:
Yes good point. I'm planning to refactor everything using a TestResource. Would it be ok if I do that in the end after all the PRs have been merged? Otherwise it's going to get difficult rebasing every single PRs as that's going to create a lot of conflicts.